cli_tools/common/distro/distro.go

```diff
@@ -0,0 +1,179 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package distro
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// Release encapsulates product and version information
+// about a Linux or Windows release.
+type Release interface {
+	ImportCompatible(other Release) bool
+	AsGcloudArg() string
+}
+
+type release struct {
+	distro  string
+	variant string
+	major   string
+	minor   string
+}
+
+// AsGcloudArg returns a string representation using the format
+// that we use with the `--os` flag.
+func (l release) AsGcloudArg() string {
+	switch l.distro {
+	case "ubuntu":
+		return fmt.Sprintf("%s-%s%s", l.distro, l.major, l.minor)
+	case "sles":
+		if l.variant != "" {
+			return fmt.Sprintf("%s-%s-%s", l.distro, l.variant, l.major)
+		} else {
+			return fmt.Sprintf("%s-%s", l.distro, l.major)
+		}
+	default:
+		return fmt.Sprintf("%s-%s", l.distro, l.major)
+	}
+}
+
+// ImportCompatible uses product-specific information
+// to determine whether two versions are compatible for import.
+func (l release) ImportCompatible(other Release) bool {
+	realOther, ok := other.(release)
+	if !ok {
+		return false
+	}
+
+	if l.distro == "ubuntu" {
```

Comment:
extract OS names to constants as they are reused

Reply:
done